pom.xml

```diff
@@ -411,6 +417,31 @@
                 </exclusion>
             </exclusions>
         </dependency>
+
+        <!-- Java native HTTP client extensions -->
+        <dependency>
+            <groupId>com.github.mizosoft.methanol</groupId>
+            <artifactId>methanol</artifactId>
```

Comment:
As discussed previously, I'd prefer if you use spring's webclient or resttemplate, even if this is in the "ext" code. If for no other reason not to add unnecessary dependencies.

Reply:
Replaced with WebClient. There was also RestClient available but opted for WebClient as the effective difference is neglible and it would be yet another wrapper-of-a-wrapper. See official docs: https://docs.spring.io/spring-boot/reference/io/rest-client.html